libs/core-ui/src/lib/Cohort/ManualCohortManagement/CohortEditorFilter.tsx

```diff
@@ -42,47 +51,83 @@ export interface ICohortEditorFilterProps {
   setComparison(ev?: React.FormEvent<IComboBox>, item?: IComboBoxOption): void;
   setNumericValue(
     delta: number,
-    column: IJointMeta,
     index: number,
-    stringVal: string
+    stringVal: string,
+    range?: INumericRange
   ): string | void;
   saveState(index?: number): void;
   cancelFilter(): void;
 }
+const maxLength = 18;
 export class CohortEditorFilter extends React.Component<ICohortEditorFilterProps> {
-  private readonly dataArray: IComboBoxOption[] = new Array(
-    this.props.jointDataset.datasetFeatureCount
-  )
-    .fill(0)
-    .map((_, index) => {
-      const key = JointDataset.DataLabelRoot + index.toString();
-      return {
-        key,
-        text: this.props.jointDataset.metaDict[key].abbridgedLabel
-      };
-    });
+  private readonly dataArray: IComboBoxOption[] = this.props.isRefactorFlightOn
+    ? this.props.dataset.feature_names.map((featureName, index) => {
+        return {
+          index,
+          key: featureName,
+          text:
+            featureName.length <= maxLength
+              ? featureName
+              : `${featureName.slice(0, maxLength)}...`
+        };
```

Comment:
nit: you could use lodash truncate which might be a little more readable here but I"m not sure if lodash is included already in the package